junit-platform-launcher/src/main/java/org/junit/platform/launcher/listeners/MutableTestExecutionSummary.java

```diff
@@ -192,13 +194,13 @@ public void printFailuresTo(PrintWriter writer) {
 
 	@Override
 	public List<Failure> getFailures() {
-		return new ArrayList<>(failures);
+		return Collections.unmodifiableList(new ArrayList<>(failures));
```

Comment:
I'm not so sure about that, because it would change the behaviour of `getFailures()` to not only be unmodifiable but also a _view_ rather than a copy, and I don't know if that behaviour is relied upon. (But then again, I don't know if `getFailures()` being mutable is relied upon as well...)